shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -447,3 +451,64 @@ - (void)clipPath:(const SkPath&)path matrix:(const SkMatrix&)matrix {
 }
 
 @end
+
+@interface FlutterClippingMaskViewPool ()
+
+@property(assign, nonatomic) NSUInteger capacity;
+@property(retain, nonatomic) NSMutableArray<FlutterClippingMaskView*>* pool;
+// The index points to the first available FlutterClippingMaskView in the `pool`.
+@property(assign, nonatomic) NSUInteger availableIndex;
+
+@end
+
+@implementation FlutterClippingMaskViewPool : NSObject
+
+- (instancetype)initWithCapacity:(NSInteger)capacity {
+  if (self = [super init]) {
+    _pool = [[NSMutableArray alloc] initWithCapacity:capacity];
+    _capacity = capacity;
+    _availableIndex = 0;
+  }
+  return self;
+}
+
+- (FlutterClippingMaskView*)getMaskViewWithFrame:(CGRect)frame {
+  FML_DCHECK(self.availableIndex <= self.capacity);
+  FlutterClippingMaskView* maskView;
+  if (self.availableIndex == self.capacity) {
```

Comment:
it looks like the `capacity` is not updated? should it be increased when you add a new one?

Reply:
we can probably just compare with `pool.count`, and do not keep this capacity ivar?

Reply:
The capacity is a static limit set for the pool. My intention is to limit the memory used by the pool.

Reply:
gotcha. I mistakenly thought that this capacity increases like an NSMutableArray.

Reply:
Ah, yeah it could be confusing, I added a comment.